snapshots/windows/windows.go

```diff
@@ -457,12 +488,47 @@ func (s *snapshotter) createUVMScratchLayer(ctx context.Context, snDir string, p
 		return errors.Wrapf(err, "failed to find UtilityVM directory in base layer %q", baseLayer)
 	}
 
-	templateDiffDisk := filepath.Join(uvmPath, "SystemTemplate.vhdx")
+	var (
+		newDisks         = sizeGB > 0 && sizeGB != 10
+		templateBase     = filepath.Join(uvmPath, "SystemTemplateBase.vhdx")
+		templateDiffDisk = filepath.Join(uvmPath, "SystemTemplate.vhdx")
+		bcdName          = fmt.Sprintf("BCD_%d", sizeGB)
+		bcdCache         = filepath.Join(uvmPath, "bcdcache")
+		bcdDestination   = filepath.Join(bcdCache, bcdName)
+		bcdSource        = filepath.Join(uvmPath, uvmBCDPath)
+	)
 
-	// Check if SystemTemplate disk doesn't exist for some reason (this should be made during the unpacking
-	// of the base layer).
-	if _, err := os.Stat(templateDiffDisk); os.IsNotExist(err) {
-		return fmt.Errorf("%q does not exist in Utility VM image", templateDiffDisk)
+	// If the cache doesn't exist this is the first time this image is being used to
+	// create a vm isolated container. Cache the original BCD entry.
+	if _, err := os.Stat(bcdCache); os.IsNotExist(err) {
+		if err := os.MkdirAll(bcdCache, 0777); err != nil {
+			return errors.Wrap(err, "failed to make BCD cache")
+		}
+		if err := copyFile(bcdSource, filepath.Join(bcdCache, fmt.Sprintf("BCD_%d", 10))); err != nil {
+			return errors.Wrap(err, "failed to copy bcd entry")
+		}
+	}
+
+	if newDisks {
+		templateBase = filepath.Join(uvmPath, fmt.Sprintf("SystemTemplateBase_%d.vhdx", sizeGB))
+		templateDiffDisk = filepath.Join(uvmPath, fmt.Sprintf("SystemTemplate_%d.vhdx", sizeGB))
+	}
+
+	if _, err := os.Stat(templateDiffDisk); os.IsNotExist(err) && newDisks {
+		if err := computestorage.SetupUtilityVMBaseLayer(ctx, uvmPath, templateBase, templateDiffDisk, uint64(sizeGB)); err != nil {
+			return errors.Wrap(err, "failed to create UVM scratch vhdx")
+		}
+		if err := copyFile(bcdSource, bcdDestination); err != nil {
```

Comment:
Shouldn't the BCD file be copied with the new name like `fmt.Sprintf("BCD_%d", sizeGB)`?

Reply:
It should be, `bcdDestination` is set to `BCD_%d` with lines 495-497 in this file although it's confusing with all of the bcd____ variables 😆